src/lib/Metadata/Metadata.js

```diff
@@ -1,134 +1,135 @@
-/*!
- * Copyright(c) ConsenSys Software Inc.
- * Copyright(c) https://consensys.net/
- * MIT Licensed
- */
+import { errorLogger, ERROR_LOG } from '../error/handler.js';
+import {
+  contractMetadataSchema,
+  freeMetadataSchema,
+  tokenMetadataSchema,
+} from './metadata.schema.js';
 
-import { readFileSync } from 'fs';
+export default class Metadata {
+  /**
+   * Create token metadata
+   * @param {string} name Name of the token
+   * @param {string} description description of the token
+   * @param {string} image tokenURI
+   * (link to a JSON file describing the token's metadata)
+   * @param {Array} attributes The attributes of the token
+   * @param {string} animation_url Link to the animation such as music, video
+   * @returns string
+   */
+  static OpenSeaTokenLevelStandard({
+    name,
+    description,
+    image,
+    attributes,
+    external_url = null,
+    animation_url = null,
+    background_color = null,
+    youtube_url = null,
+  }) {
+    const tokenMetadata = {
+      name,
+      description,
+      image,
+      attributes,
+    };
 
-import IPFS from '../../services/ipfsService.js';
-import { ERROR_LOG, errorLogger } from '../error/handler.js';
+    if (animation_url) {
+      tokenMetadata.animation_url = animation_url;
+    }
 
-export default class Metadata {
-  IPFS_BASE_URL = 'https://ipfs.io/ipfs/';
+    if (background_color) {
+      tokenMetadata.background_color = background_color;
+    }
 
-  constructor({ projectId, projectSecret }) {
-    if (!projectId) {
-      throw new Error(
-        errorLogger({
-          location: ERROR_LOG.location.Metadata_constructor,
-          message: ERROR_LOG.message.no_infura_projectID_supplied,
-        }),
-      );
+    if (external_url) {
+      tokenMetadata.external_url = external_url;
+    }
+
+    if (youtube_url) {
+      tokenMetadata.youtube_url = youtube_url;
     }
 
-    if (!projectSecret) {
+    const result = tokenMetadataSchema.validate(tokenMetadata);
+
+    if (result.error) {
       throw new Error(
         errorLogger({
-          location: ERROR_LOG.location.Metadata_constructor,
-          message: ERROR_LOG.message.no_infura_projectSecret_supplied,
+          location: ERROR_LOG.location.Metadata_token_creation,
+          message: result.error.details[0].message,
         }),
       );
     }
 
-    this.client = new IPFS({
-      projectId,
-      projectSecret,
-    });
+    return JSON.stringify(tokenMetadata);
   }
 
-  createTokenURI(metadataInput) {
-    return this.upload({
-      metadataObject: Metadata.parseInput(metadataInput),
-      isContract: false,
-    });
-  }
+  /**
+   * Create contract metadata
+   * @param {string} name Name of the contract
+   * @param {string} description description of the contract
+   * @param {string} image contractURI
```

Comment:
Not contractURI, but link to the contract level image

See https://docs.opensea.io/docs/contract-level-metadata